packages/apollo-federation/src/composition/utils.ts

```diff
@@ -360,6 +352,10 @@ export function diffTypeNodes(
     [fieldName: string]: string[];
   } = Object.create(null);
 
+  const unionTypesDiff: {
```

Comment:
Union value type support is a great add but should be done in a separate PR then this one.

Reply:
Anything in particular that makes unions out of scope? I believe accounting for them while diffing potential value types may be the only additional complexity they bring to this PR.